tests/helpers.py

```diff
@@ -930,3 +930,67 @@ def delete_deploymentconfig(pod_obj):
     """
     dc_ocp_obj = ocp.OCP(kind=constants.DEPLOYMENTCONFIG)
     dc_ocp_obj.delete(resource_name=pod_obj.get_labels().get('name'))
+
+
+def wait_for_resource_count_change(
+    func_to_use, previous_num, namespace, change_type='increase',
+    min_difference=1, timeout=20, interval=2
+):
+    """
+    Wait for a change in total count of PVC or pod
+
+    Args:
+        func_to_use (function): Function to be used to fetch resource info
+            Supported functions: pod.get_all_pvcs(), pod.get_all_pods()
+        previous_num (int): Previous number of pods/PVCs for comparison
+        namespace (str): Name of the namespace
+        change_type (str): Type of change to check. Accepted values are
+            'increase' and 'decrease'. Default is 'increase'.
+        min_difference (int): Minimum required difference in PVC/pod count
+        timeout (int): Maximum wait time in seconds
+        interval (int): Time in seconds to wait between consecutive checks
+
+    Returns:
+        True if difference in count is greater than or equal to
+            'min_difference'. False in case of timeout.
+    """
+    try:
+        for sample in TimeoutSampler(
+            timeout, interval, func_to_use, namespace
+        ):
+            if func_to_use == pod.get_all_pods:
+                current_num = len(sample)
+            else:
+                current_num = len(sample['items'])
+
+            if change_type == 'increase':
+                count_diff = current_num - previous_num
+            else:
+                count_diff = previous_num - current_num
+            if count_diff >= min_difference:
+                return True
+    except TimeoutExpiredError:
+        return False
+
+
+def verify_pv_mounted_on_node(node_pv_dict):
```

Comment:
seems like a very specific function to your test.
Maybe will be better to return the mounted PVs on a node/s and to the check, in the test itself?

Reply:
I did that initially, but after a second thought this was modified like this because the need of taking the mounts list from a node will we mostly to check the existence of volume. This is helpful in most of the PVC related  tests.